### PR TITLE
Refactor master.cfg into some new submodules

### PR DESCRIPTION
**Pull request by pablogsal**
You can skip 132a472 as that only formats the source with black to keep the style a bit more consistent (while maintaining custom style in some files).

### CODE REVIEW

**Comment by pablogsal** [comment]
This is a simple refactor that just tidies up the `master.cfg` file and factors out related functionality in different files and functions for easy reuse and organisation.

File: master/custom/builders.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+# We use one scheduler per branch (as the AnyBranchScheduler is useless)
+# and one builder per worker and branch
+
+# the 'builders' list defines the Builders. Each one is configured with a
+# dictionary, using the following keys:
+#  name (required): the name used to describe this builder
+#  workername (required): which worker to use, must appear in c['bots']
+#  builddir (required): which subdirectory to run the builder in
+#  factory (required): a BuildFactory to define how the build is run
+#  periodicBuildTime (optional): if set, force a build every N seconds
```
Comment (zware) [inline master/custom/builders.py]: These two comments can probably just go away.  Neither one makes a whole lot of sense here anymore.

File: master/custom/factories.py [hunk 2]
```diff
@@ -15,7 +23,7 @@
 def regrtest_has_cleanup(branch):
     # "python -m test --cleanup" is available in Python 3.7 and newer,
     # and in Python 2.7.
-    return (branch not in ("3.4", "3.5", "3.6", CUSTOM_BRANCH_NAME))
+    return branch not in ("3.4", "3.5", "3.6", CUSTOM_BRANCH_NAME)
```
Comment (zware) [inline master/custom/factories.py]: These branches are well out of date :)

**Comment by pablogsal** [comment]
@zware I have rebased this PR on top of the other two. This looks a bit complex, but in reality is just moving around pieces of code to new modules/functions and run black.

**Comment by pablogsal** [comment]
@zware Addressed feedback in 67e15e7

**Comment by pablogsal** [comment]
Hummm....I don't understand why GitHub marked this as closed when is clearly merged:

https://github.com/python/buildmaster-config/commit/8657e8c610fca427d54e262bf251a5e1c9691b11

**Comment by pablogsal** [comment]
I just deployed this on the server! Thanks for the reviews, @zware!

**Pull request closed without merging**